GPT/beta-commands/gpt-function-calling.py

```diff
@@ -0,0 +1,161 @@
+            {
+                "type": "function",
+                "function": {
+                    "name": "notify",
+                    "description": "notify(str: string) - this notifies the user using a popup notification",
+                    "parameters": {
+                        "type": "object",
+                        "properties": {
+                            "str": {
+                                "type": "string",
+                                "description": "The text to notify",
+                            }
+                        },
+                        "required": ["str"],
+                    },
+                },
+            },
+            {
+                "type": "function",
+                "function": {
+                    "name": "display",
+                    "description": "display(str: string) - DEFAULT - this displays the response to the user. Use this for all informational text aside from notifications. Use this instead of returning content in the response.",
+                    "parameters": {
+                        "type": "object",
+                        "properties": {
+                            "str": {
+                                "type": "string",
+                                "description": "The text to display",
+                            }
+                        },
+                        "required": ["str"],
+                    },
+                },
+            },
+            {
+                "type": "function",
+                "function": {
+                    "name": "search_for_command",
+                    "description": "search_for_command(str: string) - this searches for a command in the VSCode command palette. If I ask you to do something, please use this command to search for an appropriate command.",
+                    "parameters": {
+                        "type": "object",
+                        "properties": {
+                            "str": {
+                                "type": "string",
+                                "description": "The command to search for",
+                            }
+                        },
+                        "required": ["str"],
+                    },
+                },
+            }
+        ],
+        "max_tokens": 2024,
+        "temperature": 0.6,
+        "n": 1,
+        "stop": None,
+        "model": settings.get("user.openai_model"),
+    }
+
+
+    response = requests.post(url, headers=headers, data=json.dumps(data))
+
+    message = response.json()["choices"][0]["message"]
+    if response.status_code == 200:
+        notify("GPT Task Completed")
+        print(response.json())
+        process_function_calls(insert_response, message)
+
+        content = (message["content"] or "").strip()
+        if len(content) != 0:
+            display_response(content)
+
+    else:
+        notify("GPT Failure: Check API Key, Model, or Prompt")
+        print(response.json())
```

Comment:
we should prob throw an error here to stop execution of further talon commands if we support chaining in the future / they remove anchors

Reply:
Changed. I think you're right the only thing I don't like about throwing an error is the fact we get a stack trace that we don't really need.

Reply:
I don't think a stack trace hurts. Might be helpful for user to know where error is happening